compiler/src/iree/compiler/ConstEval/Runtime.cpp

```diff
@@ -107,65 +280,35 @@ LogicalResult CompiledBinary::invokeNullary(Location loc, StringRef name,
-
-  if (auto status =
-          iree_vm_invoke(context.get(), function, IREE_VM_INVOCATION_FLAG_NONE,
-                         /*policy=*/nullptr, inputs.get(), outputs.get(),
-                         iree_allocator_system())) {
-    std::string message;
-    message.resize(512);
-    iree_host_size_t buffer_length;
-    if (!iree_status_format(status, message.size(), &message[0],
-                            &buffer_length)) {
-      message.resize(buffer_length + 1);
-      iree_status_format(status, message.size(), &message[0], &buffer_length);
-    }
-    message.resize(buffer_length);
-    iree_status_ignore(status);
-    return emitError(loc) << "internal error evaling constant: " << message;
-  }
+  return handleRuntimeError(loc, iree_vm_invoke(binary.context.get(), function,
+                                                IREE_VM_INVOCATION_FLAG_NONE,
+                                                /*policy=*/nullptr,
+                                                inputs.get(), outputs.get(),
+                                                iree_allocator_system()));
+}
 
-  if (failed(callback(outputs.get()))) {
+LogicalResult FunctionCall::getResultAsAttr(Location loc, size_t index,
+                                            TypedAttr &outAttr) {
+  iree_vm_variant_t variant = iree_vm_variant_empty();
+  if (failed(handleRuntimeError(loc, iree_vm_list_get_variant_assign(
+                                         outputs.get(), index, &variant))))
     return failure();
-  }
-  return success();
-}
 
-Attribute CompiledBinary::invokeNullaryAsAttribute(Location loc,
-                                                   StringRef name) {
-  Attribute result;
-  if (failed(invokeNullary(
-          loc, name, [&](iree_vm_list_t *outputs) -> LogicalResult {
-            if (iree_vm_list_size(outputs) != 1) {
-              return emitError(loc) << "expected 1 result for func " << name
-                                    << " got " << iree_vm_list_size(outputs);
-            }
-            iree_vm_variant_t variant = iree_vm_variant_empty();
-            IREE_CHECK_OK(
-                iree_vm_list_get_variant_assign(outputs, 0, &variant));
-            result = convertVariantToAttribute(loc, variant);
-            return success(result != nullptr);
-          }))) {
-    return nullptr;
-  }
+  outAttr = binary.convertVariantToAttribute(loc, variant);
+  if (!outAttr)
+    return failure();
 
-  return result;
+  return success();
 }
 
 bool CompiledBinary::isSupportedResultType(Type type) {
   // TODO(laurenzo): Not currently supported. VMVX would need to support these
   // and today it doesn't. We could use alternative backends (LLVM CPU/etc) if
   // we wanted to handle f64, but f16 and bf16 often need special hardware.
-  if (llvm::isa<Float16Type>(type) || llvm::isa<BFloat16Type>(type) ||
-      llvm::isa<Float64Type>(type)) {
-    return false;
-  }
+  // if (llvm::isa<Float16Type>(type) || llvm::isa<BFloat16Type>(type) ||
```

Comment:
intended?

Reply:
Obsolete and forgot to delete the whole method.